core/templates/domain/objects/NumberWithUnitsObjectFactorySpec.ts

```diff
@@ -452,5 +452,35 @@ describe('NumberWithUnitsObjectFactory', () => {
         ).toThrowError('Number with type fraction cannot have a real part.');
       }
     );
+
+    it('should convert units to their canonical form', () => {
```

Comment:
I think you need a lot more tests for this, e.g. currency units, invalid units, units unchanged by the function, etc. Maybe have a specific test for sorting as well instead of having that only implicitly in this one.

Reply:
This function will not be called with invalid units, as the input is validated before submitting the answer.

Done, added more tests.

Reply:
This function will not be called with invalid units, as the input is validated before submitting the answer.

Done, added more tests.